test/benchmarks/index.ts

```diff
@@ -0,0 +1,210 @@
+import { readFile, writeFile } from 'fs/promises';
+
+import { getCanisterId } from '../../dfx';
+// @ts-ignore We would have to add "resolveJsonModule": true to every test tsconfig.json file
+import { version } from '../../package.json';
+import { jsonParse, jsonStringify } from '../../src/lib/stable';
+import { createActor } from './actor';
+
+type Benchmark = [bigint, { method_name: string; instructions: bigint }];
+
+type BenchmarksJson = {
+    current: {
+        version: string;
+        benchmarks: Benchmark[];
+    };
+    previous: {
+        version: string;
+        benchmarks: Benchmark[];
+    };
+};
+
+export async function runBenchmarksForCanister(
+    canisterName: string
+): Promise<void> {
+    const canisterId = getCanisterId(canisterName);
+    const actor = await createActor(canisterId, 'default'); // TODO do we need to make an identity for benchmarks just like we do for the uploader?
```

Comment:
@lastmjs Thoughts?

Reply:
It should just use the current dfx identity right?

Reply:
done